src/main/java/com/inventoryservice/service/CategoryService.java

```diff
@@ -0,0 +1,68 @@
+package com.inventoryservice.service;
+
+import com.inventoryservice.model.Category;
+import com.inventoryservice.model.SubCategory;
+import com.inventoryservice.repository.CategoryRepository;
+import com.inventoryservice.repository.SubCategoryRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class CategoryService {
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @Autowired
+    private SubCategoryRepository subcategoryRepository;
+
+    /*The method used to create the Category*/
+    public Category createCategory(Category category) {
+        return categoryRepository.save(category);
+    }
+
+    /*The method used to list of  the Category*/
+    public List<Category> getAllCategories() {
+        return categoryRepository.findAll();
+    }
+
+    /*
+     *The method used to  get the Category by Using  categoryId.*/
+    public Category findCategoryById(int categoryId) {
+        Optional<Category> category = categoryRepository.findById(categoryId);
+        return category.orElse(null);
+    }
+
+    /*
+   The method used to  delete the Category.
+   This method check Category is present or not   .
+   */
+    public Category deleteCategory(int categoryId) {
+        Category category = categoryRepository.findById(categoryId).get();
+        if (!category.isArchived()) ;
+        {
```

Comment:
format

Reply:
Do we need to check again if(category.isArchieved)

Reply:
> Do we need to check again if(category.isArchieved)

No we do not need to check if(category.isArchieved) as it is already verify in controller method.
Code formatting completed.